study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/BrainPrequalV2.java

```diff
@@ -0,0 +1,275 @@
+        }
+        this.dataCfg = ConfigFactory.parseFile(file).resolveWith(varsCfg);
+    }
+
+    @Override
+    public void run(Handle handle) {
+        LOG.info("Executing BrainPrequalV2 task...");
+
+        String activityCode = dataCfg.getString("activityCode");
+        String versionTag = dataCfg.getString("versionTag");
+        UserDto adminUser = handle.attach(JdbiUser.class).findByUserGuid(cfg.getString("adminUser.guid"));
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(cfg.getString("study.guid"));
+        Instant timestamp = Instant.now();
+        String studyGuid = studyDto.getGuid();
+        long studyId = studyDto.getId();
+        long activityId = ActivityBuilder.findActivityId(handle, studyId, activityCode);
+
+        ActivityDto activityDto = handle.attach(JdbiActivity.class)
+                .findActivityByStudyGuidAndCode(studyGuid, activityCode)
+                .orElseThrow(() -> new DDPException("Could not find id for activity " + activityCode + " and study id " + studyGuid));
+
+        String reason = String.format(
+                "Update activity with studyGuid=%s activityCode=%s to versionTag=%s",
+                studyGuid, activityCode, versionTag);
+        RevisionMetadata meta = new RevisionMetadata(timestamp.toEpochMilli(), adminUser.getUserId(), reason);
+        SqlHelper helper = handle.attach(SqlHelper.class);
+        String selfOptionText = "I have been diagnosed with brain cancer.";
+        String promptText = "First, how will you be participating in the Brain Tumor Project? ";
+
+        //load new activity def/conf
+        ActivityDef def = GsonUtil.standardGson().fromJson(ConfigUtil.toJson(dataCfg), ActivityDef.class);
+
+        //load current activity def from db
+        JdbiRevision jdbiRevision = handle.attach(JdbiRevision.class);
+        ActivityDao activityDao = handle.attach(ActivityDao.class);
+        ActivityVersionDto currentActivityVerDto = activityDao.getJdbiActivityVersion().findByActivityCodeAndVersionTag(
+                studyId, activityCode, "v1").get();
+        ActivityDef currActivityDef = activityDao.findDefByDtoAndVersion(activityDto, currentActivityVerDto);
+        FormActivityDef currFormActivityDef = (FormActivityDef)currActivityDef;
+
+        //change version
+        ActivityVersionDto activityVersionDto = activityDao.changeVersion(activityId, versionTag, meta);
+
+        //remove study activity title
+        helper.updateActivityTitle(activityDto.getActivityId(), "");
+
+        String templateVarName = "brain_prequal_contact_title";
+        long templateId = helper.findTemplateIdByVariableName(templateVarName);
+        long templateRevId = helper.findTemplateRevisionId(templateId);
+        long newTemplateRevId = jdbiRevision.copyAndTerminate(templateRevId, meta);
+        //may be look for template brain_prequal_contact_title and disable rather than these sql updates !!
```

Comment:
I agree. I think we should find the row in `i18n_template_substitution`, disable it, and add a new row with new revision. Updating the value directly like this will change the text for all users, even ones with v1.

Reply:
since its just prequal text, thought it might be ok !

Reply:
I'll still recommend doing it to minimize tangling things between the two versions.